server/aws-lsp-codewhisperer/script/generateServiceClient.ts

```diff
@@ -0,0 +1,242 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as child_process from 'child_process'
+import * as fs from 'fs'
+import * as path from 'path'
+
+const repoRoot = path.join(process.cwd()) // root/packages/toolkit -> root/
+/**
+ * This script uses the AWS JS SDK v2 to generate service clients where the client definition is contained within
+ * this repo. Client definitions are added at the bottom of this script.
+ */
+
+interface ServiceClientDefinition {
+    serviceName: string
+    serviceJsonPath: string
+}
+
+async function generateServiceClients(serviceClientDefinitions: ServiceClientDefinition[]): Promise<void> {
+    const tempJsSdkPath = path.join(repoRoot, 'node_modules', '.zzz-awssdk2')
+    console.log(`Temp JS SDK Repo location: ${tempJsSdkPath}`)
+    console.log('Service Clients to Generate: ', serviceClientDefinitions.map(x => x.serviceName).join(', '))
+
+    await cloneJsSdk(tempJsSdkPath)
+    await insertServiceClientsIntoJsSdk(tempJsSdkPath, serviceClientDefinitions)
+    await runTypingsGenerator(tempJsSdkPath)
+    await integrateServiceClients(tempJsSdkPath, serviceClientDefinitions)
+
+    console.log('Done generating service client(s)')
+}
+
+/** When cloning aws-sdk-js, we want to pull the version actually used in package-lock.json. */
+function getJsSdkVersion(): string {
+    const json = fs.readFileSync(path.resolve(repoRoot, 'package-lock.json')).toString()
+    const packageLock = JSON.parse(json)
+
+    return packageLock['packages']['node_modules/aws-sdk']['version']
+}
+
+async function cloneJsSdk(dir: string): Promise<void> {
```

Comment:
Why is it needed to pull github repo in order to build the clients? Why can't package installed in node_modules can be used directly?

Reply:
> Why is it needed to pull github repo in order to build the clients? Why can't package installed in node_modules can be used directly?

It probably could. Using git (in aws-toolkit-vscode's build) might be some tech debt that never got cleaned up. Just needs some human attention.